src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -146,7 +146,7 @@ public Task ApplyAsync(HttpContext httpContext, CandidateSet candidates)
                 for (var j = 0; j < metadata.HttpMethods.Count; j++)
                 {
                     var candidateMethod = metadata.HttpMethods[j];
-                    if (!string.Equals(httpMethod, candidateMethod, StringComparison.OrdinalIgnoreCase))
+                    if (!(object.ReferenceEquals(httpMethod, candidateMethod) || StringComparer.OrdinalIgnoreCase.Equals(httpMethod, candidateMethod)))
```

Comment:
Is there a reason why `StringComparer.OrdinalIgnoreCase.Equals(httpMethod, candidateMethod)` is better than `string.Equals(httpMethod, candidateMethod, StringComparison.OrdinalIgnoreCase)`?